### PR TITLE
[18187] VVC Video Appointment Details

### PR DESCRIPTION
**Pull request by LanceGundersen**
## Description
This PR adds updated VVC appointment details page styling and interactions.

How to test

1. Make sure the `va_online_scheduling_homepage_refresh` flag is turned on
2. On the appointments page `http://localhost:3001/health-care/schedule-view-va-appointments/appointments/` Click on each video appointment type: [Atlas, VA Location, At Home, VA Device]
3. Verify each matches the view in the specs

## Testing done
Local and unit

## Screenshots
image
image
image
image
image
image

## Acceptance criteria
- [ ] Changes are behind the va_online_scheduling_homepage_refresh feature flag
- [ ] When user clicks/taps a VVC appointment card, appointment details open on a new page
- [ ] Appointment detail page has a unique URL
- [ ] DESIGN - All desktop, tablet, and mobile designs match the specs
Mobile confirmed - https://www.sketch.com/s/4198dca5-69c7-4769-9686-d213c900876d/a/Kv9bDLV#Inspector
Mobile expanded - https://www.sketch.com/s/4198dca5-69c7-4769-9686-d213c900876d/a/wL9poM9#Inspector
All fields template - https://www.sketch.com/s/4198dca5-69c7-4769-9686-d213c900876d/a/ELn2Ldx#Inspector

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/appointment-list/components/confirmedAppointmentDetailsPage/AtlasLocation.jsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  const name = `ATLAS facility in ${city}, ${state}`;
+  const facility = {
+    name,
+    address,
+  };
+
+  return (
+    <div>
+      <FacilityAddress facility={facility} showDirectionsLink />
+      <div className="vads-u-font-weight--bold vads-u-margin-top--2">
```
Comment (jbalboni) [inline src/applications/vaos/appointment-list/components/confirmedAppointmentDetailsPage/AtlasLocation.jsx:22]: I think I updated this to be a heading in the old component (after you started this PR), so we may want to carry over that change.

File: src/applications/vaos/appointment-list/components/confirmedAppointmentDetailsPage/index.jsx [hunk 2]
```diff
@@ -100,15 +120,30 @@ function ConfirmedAppointmentDetailsPage({
...
+            header={header}
+            facility={facility}
+            appointment={appointment}
+          />
+        </>
+      )}
+
       {!!facility &&
-        !isExpressCare && (
+        !isExpressCare &&
```
[the remark this answers is not in the document]
Comment (LanceGundersen) [reply src/applications/vaos/appointment-list/components/confirmedAppointmentDetailsPage/index.jsx:139]: Removed

File: src/applications/vaos/appointment-list/components/confirmedAppointmentDetailsPage/AppointmentDateTime.jsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  return stripDST(tzAbbr);
+}
+
+export function getAppointmentTimezoneDescription(timezone, facilityId) {
+  const abbr = getAppointmentTimezoneAbbreviation(timezone, facilityId);
+
+  return getTimezoneDescFromAbbr(abbr);
+}
+
+export default function AppointmentDateTime({
```
[the remark this answers is not in the document]
Comment (LanceGundersen) [reply src/applications/vaos/appointment-list/components/confirmedAppointmentDetailsPage/AppointmentDateTime.jsx:23]: Makes sense, removed.

[the remark this answers is not in the document]
Comment (LanceGundersen) [reply src/applications/vaos/appointment-list/components/confirmedAppointmentDetailsPage/AppointmentDateTime.jsx:23]: Actually, @jbalboni per the specs the timezone is added 
image

File: src/applications/vaos/appointment-list/components/confirmedAppointmentDetailsPage/AppointmentInstructions.jsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+
+export default function AppointmentInstructions({
+  instructions,
+  isHomepageRefresh,
```
[the remark this answers is not in the document]
Comment (LanceGundersen) [reply src/applications/vaos/appointment-list/components/confirmedAppointmentDetailsPage/AppointmentInstructions.jsx:5]: Makes sense removed that bit of logic

File: src/applications/vaos/appointment-list/components/confirmedAppointmentDetailsPage/index.jsx [hunk 5]
```diff
@@ -33,7 +38,21 @@ function formatAppointmentDate(date) {
...
+  } else if (isAtlasLocation(appointment)) {
+    return 'VA Video Connect at an ATLAS location';
+  } else if (isVideoHome(appointment)) {
+    return 'VA Video Connect at home';
+  } else {
+    return 'VA Appointment';
+  }
+}
+
+function Index({
```
[the remark this answers is not in the document]
Comment (LanceGundersen) [reply src/applications/vaos/appointment-list/components/confirmedAppointmentDetailsPage/index.jsx:55]: Done ✅

File: src/applications/vaos/appointment-list/components/confirmedAppointmentDetailsPage/AdditionalInfoRow.jsx [hunk 6]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import ReactCSSTransitionGroup from 'react-transition-group/CSSTransitionGroup';
+
+export default function AdditionalInfoRow({
```
[the remark this answers is not in the document]
Comment (LanceGundersen) [reply src/applications/vaos/appointment-list/components/confirmedAppointmentDetailsPage/AdditionalInfoRow.jsx:4]: Nice, plugged it in.

File: src/applications/vaos/appointment-list/components/confirmedAppointmentDetailsPage/index.jsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (LanceGundersen) [reply src/applications/vaos/appointment-list/components/confirmedAppointmentDetailsPage/index.jsx:55]: Done

File: src/applications/vaos/appointment-list/components/confirmedAppointmentDetailsPage/index.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (LanceGundersen) [reply src/applications/vaos/appointment-list/components/confirmedAppointmentDetailsPage/index.jsx:139]: removed

File: src/applications/vaos/appointment-list/components/confirmedAppointmentDetailsPage/VideoVisitSection.jsx [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+          <a href="#">Print</a>
+        </div>
+        <AlertBox
+          status={ALERT_TYPE.INFO}
+          className="vads-u-display--block"
+          backgroundOnly
+        >
+          Contact this facility if you need to reschedule or cancel your
+          appointment.
+          {!isAtlas &&
```
[the remark this answers is not in the document]
Comment (LanceGundersen) [reply src/applications/vaos/appointment-list/components/confirmedAppointmentDetailsPage/VideoVisitSection.jsx:176]: Per spec https://www.sketch.com/s/4198dca5-69c7-4769-9686-d213c900876d/a/L00Lbgl

File: src/applications/vaos/appointment-list/components/confirmedAppointmentDetailsPage/AppointmentDateTime.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (jbalboni) [reply src/applications/vaos/appointment-list/components/confirmedAppointmentDetailsPage/AppointmentDateTime.jsx:23]: @LanceGundersen I mean that the `timeZone` prop is never passed. The component gets the timezone based on the facility in those helper functions in the component and displays that.

File: src/applications/vaos/appointment-list/index.jsx [hunk 8]
```diff
@@ -6,7 +6,7 @@ import PageLayout from './components/AppointmentsPage/PageLayout';
 import AppointmentsPageV2 from './components/AppointmentsPage/AppointmentsPageV2';
 import AppointmentsPage from './components/AppointmentsPage/index';
 import RequestedAppointmentDetailsPage from './components/RequestedAppointmentDetailsPage';
-import ConfirmedAppointmentDetailsPage from './components/ConfirmedAppointmentDetailsPage';
+import ConfirmedAppointmentDetailsPage from './components/confirmedAppointmentDetailsPage';
```
[the remark this answers is not in the document]
Comment (LanceGundersen) [reply src/applications/vaos/appointment-list/index.jsx:9]: Done

File: src/applications/vaos/appointment-list/components/confirmedAppointmentDetailsPage/AtlasLocation.jsx (hunk 1 above)
Comment (LanceGundersen) [reply src/applications/vaos/appointment-list/components/confirmedAppointmentDetailsPage/AtlasLocation.jsx:22]: I see it now, fixing.

File: src/applications/vaos/appointment-list/components/confirmedAppointmentDetailsPage/index.jsx [hunk 9]
```diff
@@ -100,13 +120,28 @@ function ConfirmedAppointmentDetailsPage({
         />
       </h1>
 
+      {isVideo && (
+        <>
+          <div className="vads-u-font-size--sm vads-u-font-family--sans">
+            <span>
+              <strong>{header}</strong>
```
[the remark this answers is not in the document]
Comment (LanceGundersen) [reply src/applications/vaos/appointment-list/components/confirmedAppointmentDetailsPage/index.jsx:127]: Fixed this and a few other spots as well

File: src/applications/vaos/appointment-list/components/confirmedAppointmentDetailsPage/VideoVisitSection.jsx [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+            !isVA && (
+              <span className="vads-u-display--block vads-u-margin-top--2">
+                {name}
```
[the remark this answers is not in the document]
Comment (LanceGundersen) [reply src/applications/vaos/appointment-list/components/confirmedAppointmentDetailsPage/VideoVisitSection.jsx:179]: Fixed, makes sense to me!

File: src/applications/vaos/appointment-list/components/confirmedAppointmentDetailsPage/index.jsx [hunk 11]
```diff
@@ -100,15 +120,30 @@ function ConfirmedAppointmentDetailsPage({
...
+            </span>
+          </div>
+          <VideoVisitSection
+            header={header}
+            facility={facility}
```
[the remark this answers is not in the document]
Comment (LanceGundersen) [reply src/applications/vaos/appointment-list/components/confirmedAppointmentDetailsPage/index.jsx:132]: Fixed

File: src/applications/vaos/tests/mocks/v0.js [hunk 12]
```diff
@@ -112,7 +112,30 @@ export function getVideoAppointmentMock() {
               },
             },
           ],
-          providers: [],
+          providers: [
```
[the remark this answers is not in the document]
Comment (LanceGundersen) [reply src/applications/vaos/tests/mocks/v0.js:115]: Fixed

File: src/applications/vaos/appointment-list/index.jsx (hunk 8 above)
Comment (vbahinwillit) [inline src/applications/vaos/appointment-list/index.jsx:9]: Why the lowercase "confirmedAppointmentDetailsPage"?

**Review by vbahinwillit — COMMENTED** [review]
LGTM. I see there are some merge conflicts that need to be resolved and I only had one comment about on the case of a renamed file.

File: src/applications/vaos/appointment-list/index.jsx (hunk 8 above)
Comment (LanceGundersen) [reply src/applications/vaos/appointment-list/index.jsx]: Fixed

File: src/applications/vaos/appointment-list/components/confirmedAppointmentDetailsPage/AppointmentDateTime.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (LanceGundersen) [reply src/applications/vaos/appointment-list/components/confirmedAppointmentDetailsPage/AppointmentDateTime.jsx]: Fixed

File: src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/AdditionalInfoRow.jsx [hunk 13]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import ReactCSSTransitionGroup from 'react-transition-group/CSSTransitionGroup';
+
+export default function AdditionalInfoRow({
```
Comment (jbalboni) [inline src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/AdditionalInfoRow.jsx:4]: I think you can delete this file

File: src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/VideoVisitSection.jsx [hunk 14]
```diff
@@ -0,0 +1,190 @@
...
+    if (atlasLocation?.address) {
+      return formatFacilityAddress(atlasLocation);
+    }
+  } else if (videoKind === VIDEO_TYPES.clinic) {
+    return facility ? formatFacilityAddress(facility) : null;
+  } else if (videoKind === VIDEO_TYPES.gfe) {
+    return 'Video conference';
+  } else if (isVideo) {
+    return 'Video conference';
+  } else if (isCommunityCare) {
```
Comment (jbalboni) [inline src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/VideoVisitSection.jsx:50]: You could remove the conditionals in here that aren't video related, but it won't break anything.

**Review by jbalboni — APPROVED** [review]
Couple of very minor cleanup things, looks good otherwise.

File: src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/VideoVisitSection.jsx (hunk 14 above)
Comment (LanceGundersen) [reply src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/VideoVisitSection.jsx]: Good catch, fixed.

File: src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/AdditionalInfoRow.jsx (hunk 13 above)
Comment (LanceGundersen) [reply src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/AdditionalInfoRow.jsx]: Done

**Pull request merged**